extensions/amp-story/1.0/amp-story.js

```diff
@@ -1089,28 +1097,39 @@ export class AmpStory extends AMP.BaseElement {
   }
 
   /**
-   * Displays the unsupported browser UI: either the publisher provided UI, or
-   * fallbacks to a generic default.
+   * If browser is supported, displays the story. Otherwise, shows either the
+   * default unsupported browser layer or the publisher fallback (if provided).
    * @param {boolean} isBrowserSupported
    * @private
    */
   onSupportedBrowserStateUpdate_(isBrowserSupported) {
-    if (isBrowserSupported) {
-      dev().error(TAG, 'No handler to exit unsupported browser state.');
-    }
-
     const fallbackEl = this.getFallback();
-
-    this.mutateElement(() => {
-      this.element.classList.add('i-amphtml-story-fallback');
-    });
-
-    // Displays the publisher provided fallback, or fallbacks to the default
-    // unsupported browser layer.
-    if (fallbackEl) {
-      this.toggleFallback(true);
+    if (isBrowserSupported) {
+      // Removes the default unsupported browser layer or throws an error
+      // if the publisher has provided their own fallback
+      if (fallbackEl) {
+        dev().error(TAG, 'No handler to exit unsupported browser state on ' +
+        'publisher provided fallback.');
+      } else {
+        //removeElement(this.unsupportedBrowserLayer_.get());
+        this.unsupportedBrowserLayer_.remove();
+        this.mutateElement(() => {
+          this.element.classList.remove('i-amphtml-story-fallback');
+        });
+        this.layoutStory_();
```

Comment:
I wonder if this can cause race conditions where removing the fallback element and class can happen before the story finishes being laid out?  Maybe we should do:

```javascript
this.layoutStory_().then(() => this.mutateElement(...))
```

This will cause some UX jank (it will feel like we didn't respond to the tap if `this.layoutStory_` takes a while), which we may also want to fix.